kie-maven-plugin/src/main/java/org/kie/maven/plugin/KieInMemoryMetaInfoBuilder.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    public KieModuleMetaInfo getKieModuleMetaInfo() {
+        KieModuleMetaInfo info = this.generateKieModuleMetaInfo();
+        //     this.trgMfs.write("META-INF/kmodule.info", info.marshallMetaInfos().getBytes(IoUtils.UTF8_CHARSET), true);
+        return info;
+    }
+
+    public MemoryFileSystem getMemoryFileSystem() {
+        return this.trgMfs;
+    }
+
+    private KieModuleMetaInfo generateKieModuleMetaInfo() {
+        Map<String, TypeMetaInfo> typeInfos = new HashMap();
+        Map<String, Set<String>> rulesPerPackage = new HashMap();
+        KieModuleModel kieModuleModel = this.kModule.getKieModuleModel();
+        Iterator var4 = kieModuleModel.getKieBaseModels().keySet().iterator();
+
+        while (var4.hasNext()) {
+            String kieBaseName = (String) var4.next();
+            KnowledgeBuilderImpl kBuilder = (KnowledgeBuilderImpl) this.kModule.getKnowledgeBuilderForKieBase(kieBaseName);
+            Map<String, PackageRegistry> pkgRegistryMap = kBuilder.getPackageRegistry();
+            KieModuleCache.KModuleCache.Builder _kmoduleCacheBuilder = this.createCacheBuilder();
+            KieModuleCache.CompilationData.Builder _compData = this.createCompilationData();
+
+            JavaDialectRuntimeData runtimeData;
+            ArrayList types;
+            for (Iterator var10 = kBuilder.getKnowledgePackages().iterator(); var10.hasNext(); this.addToCompilationData(_compData, runtimeData, types)) {
+                KiePackage kPkg = (KiePackage) var10.next();
+                PackageRegistry pkgRegistry = (PackageRegistry) pkgRegistryMap.get(kPkg.getName());
+                runtimeData = (JavaDialectRuntimeData) pkgRegistry.getDialectRuntimeRegistry().getDialectData("java");
+                types = new ArrayList();
+
+                String internalName;
+                for (Iterator var15 = kPkg.getFactTypes().iterator(); var15.hasNext(); types.add(internalName)) {
+                    FactType factType = (FactType) var15.next();
+                    Class<?> typeClass = ((ClassDefinition) factType).getDefinedClass();
+                    TypeDeclaration typeDeclaration = pkgRegistry.getPackage().getTypeDeclaration(typeClass);
+                    if (typeDeclaration != null) {
+                        typeInfos.put(typeClass.getName(), new TypeMetaInfo(typeDeclaration));
+                    }
+
+                    String className = factType.getName();
+                    internalName = className.replace('.', '/') + ".class";
+                    byte[] bytes = runtimeData.getBytecode(internalName);
+                    if (bytes != null) {
+                        this.trgMfs.write(internalName, bytes, true);
+                    }
+                }
+
+                Set<String> rules = (Set) rulesPerPackage.get(kPkg.getName());
+                if (rules == null) {
+                    rules = new HashSet();
+                }
+
+                Iterator var23 = kPkg.getRules().iterator();
```

Comment:
Sure, the KieInMEmoryMetaInfoBuilder is a copy of KieMetaInfoBuilder but with a in MemoryFS  instead of a regular FS, this class is to preserve the behaviour of the plugin, the goal of the changes are i this subtask https://issues.jboss.org/browse/GUVNOR-3124 and in the main task https://issues.jboss.org/browse/GUVNOR-3016